### PR TITLE
More mixpanel logs

### PR DESCRIPTION
**Pull request by nramirez**
Removing more console logs, and tracking everything with mixpanel

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -80,13 +87,14 @@ app.post('/new-message', (req, res) => {
       } else if (message.new_chat_participant || message.left_chat_participant) {
         return addOrRemoveParticipant(message, chatId, usernames, ref, res);
       }
-    }, error => {
-      console.log('Error reading db:', error);
-      return res.end('Error reading db:', error);
+    }, e => {
```
Comment (nramirez) [inline index.js]: let's change this, to `.catch` as well, reads more human

File: index.js [hunk 2]
```diff
@@ -106,44 +114,52 @@ const setUsernames = (message, chatId, usernames, ref, res) => {
...
-          usernames: messageManager.extractUniqueUsernames(message, usernames),
+          usernames: users,
         }).then(m => {
           telegramManager.talkToBot(chatId, 'Usernames added.')
-            .then(() => res.end('Updated message:', m))
-            .catch(err => {
-              console.log(err);
-              res.end('Error: ', err);
+            .then(() => {
+              mixpanel.track('Usernames set', { users, message });
```
Comment (nramirez) [inline index.js]: let's change this to `setUsernames` to follow the pattern of the others.

File: index.js [hunk 3]
```diff
@@ -106,44 +114,52 @@ const setUsernames = (message, chatId, usernames, ref, res) => {
...
+              res.end('Updated message:', m);
+            })
+            .catch(e => {
+              mixpanel.track('Error talkToBot new usernames', { e, message });
```
Comment (nramirez) [inline index.js]: ditto

File: index.js [hunk 4]
```diff
@@ -106,44 +114,52 @@ const setUsernames = (message, chatId, usernames, ref, res) => {
...
+              res.end('Error updating db usernames ', e);
             });
         }).catch(e => {
-          console.log('Error updating db:', e);
-          res.end('Error updating db:', e);
+          mixpanel.track('Error updating chat users', { e, chatId, message });
```
Comment (nramirez) [inline index.js]: ditto

File: index.js [hunk 5]
```diff
@@ -106,44 +114,52 @@ const setUsernames = (message, chatId, usernames, ref, res) => {
...
+          res.end('Error updating chat users', e);
         });
       } else {
         telegramManager.talkToBot(chatId, 'This action is only allowed for admins.')
-          .then(() => res.end('This action is only allowed for admins.'))
-          .catch(err => {
-            console.log(err);
-            res.end('Error: ', err);
+          .then(() => {
+            mixpanel.track('Not Admin', { chatId, message });
```
Comment (nramirez) [inline index.js]: be more specific: `setUsernames talkToBot  Admin Only`

File: index.js [hunk 6]
```diff
@@ -176,26 +194,27 @@ const clearUsernames = (message, chatId, ref, res) => {
             usernames: '',
           });
           telegramManager.talkToBot(chatId, 'All cleared.').then(() => {
-            res.end('All cleared.');
-          }, err => {
-            console.log(err);
-            res.end('Error: ', err);
+            mixpanel.track('talkToBot clearUsernames', { chatId, message });
+            return res.end('All cleared.');
+          }, e => {
```
Comment (nramirez) [inline index.js]: catch

File: index.js [hunk 7]
```diff
@@ -176,26 +194,27 @@ const clearUsernames = (message, chatId, ref, res) => {
...
-            console.log(err);
-            res.end('Error: ', err);
+          }).catch(e => {
+            res.end('Error: ', e);
           });
         }
-      }, err => {
-        console.log(err);
-        res.end('Error: ', err);
+      }, e => {
```
Comment (nramirez) [inline index.js]: catch

File: index.js [hunk 8]
```diff
@@ -215,21 +234,47 @@ const addOrRemoveParticipant = (message, chatId, usernames, ref, res) => {
     `${message.left_chat_participant.username} was removed from the list.`;
 
   telegramManager.talkToBot(chatId, msg).then(() => {
+    mixpanel.track('talkToBot addOrRemoveParticipant', { chatId, message, msg });
     return res.end(msg);
-  }, err => {
-    console.log(err);
-    return res.end('Error: ', err);
+  }, e => {
```
Comment (nramirez) [inline index.js]: catch

File: index.js [hunk 9]
```diff
@@ -215,21 +234,47 @@ const addOrRemoveParticipant = (message, chatId, usernames, ref, res) => {
...
   });
 };
 
 const readTeams = (chatId, chat, res) => {
   const teams = Object.keys(chat).filter(t => t !== 'usernames').join('\n');
   telegramManager.talkToBot(chatId, teams)
-    .then(t => res.end('Team list displayed', t))
-    .catch(e => res.end('Team list error', e));
+    .then(t => {
+      mixpanel.track('Error talkToBot', { chatId, teams });
```
Comment (nramirez) [inline index.js]: `readTeams`

File: index.js [hunk 10]
```diff
@@ -176,26 +194,27 @@ const clearUsernames = (message, chatId, ref, res) => {
...
           });
         } else {
           telegramManager.talkToBot(chatId, 'This action is only allowed for admins.').then(() => {
+            mixpanel.track('talkToBot clearUsernames Admin Only', { chatId, message });
             res.end('This action is only allowed for admins.');
-          }, err => {
-            console.log(err);
-            res.end('Error: ', err);
+          }).catch(e => {
+            res.end('Error: ', e);
```
Comment (wistcc) [inline index.js]: Forgot to add mixpanel here

**Review by wistcc — APPROVED** [review]
Great!

**Pull request merged**